src/ripple/app/ledger/Ledger.cpp

```diff
@@ -209,9 +209,9 @@ Ledger::Ledger (uint256 const& parentHash,
                 beast::Journal j)
     : mImmutable (true)
     , txMap_ (std::make_shared <SHAMap> (
-        SHAMapType::TRANSACTION, transHash, family))
-    , stateMap_ (std::make_shared <SHAMap> (
-        SHAMapType::STATE, accountHash, family))
+        SHAMapType::TRANSACTION, transHash, family, SHAMap::version{1}))
+    , stateMap_ (std::make_shared <SHAMap> (SHAMapType::STATE, accountHash,
```

Comment:
Its a violation of encapsulation / levelization that a `SHAMap` knows whether its a state map or a transaction map. Can you imagine if `vector` worked this way? Its not your fault of course, this code was already like that before your changes. But it should make you mad.